packager/media/base/media_handler_test_base.cc

```diff
@@ -249,6 +249,7 @@ std::unique_ptr<SegmentInfo> MediaHandlerTestBase::GetSegmentInfo(
   info->start_timestamp = start_timestamp;
   info->duration = duration;
   info->is_subsegment = is_subsegment;
+  info->segment_index = start_timestamp / duration + 1;
```

Comment:
Why is there a "+1" in the end? Shouldn't `segment_index` be 0 based?

Reply:
> Thanks. This is a partial review as I haven't looked at your changes on WebM and mp4.
> 
> I like your change on TS writer! Left a few comments, which probably applies to WebM and mp4 too.
> 
> Would you mind breaking this PR into multiple smaller PRs? Sorry I should have talked to you about it earlier as it is really difficult to review a gigantic PR. It would also take a long time to finish.
> 
> For example, one of the PR could be TS writer / TS segmenter.

@kqyang 
Makes sense. I will divide the pr into 
1. ChunkHandler - for sending segment index 
2. TS writer/ TS segmenter 
3. Webm
4. MP4 
5. WebVTT + rest

How does that sound?

Reply:
SGTM. Thanks.

Reply:
@kqyang Have created 2 PRs for WebM and MPEG2TS changes. Will create a PR for sending segment index and handling when these two are merged. Please review and let me know.

Reply:
Great. Thanks a lot for doing that.

Reply:
@kqyang Have created 1 PR that sends segment index from (chunking_handler, text_chunker) ->  muxer -> segmenters to generate segment with segment name based on segment index -> MpdNotifyMuxerListener -> SimpleMpdNotifier -> Representation which sets the start number based on segment index that it receives. 
Please review and let me know. Thanks.

Reply:
@sr1990 Thanks. Will take a look in the next few days.